comp/dogstatsd/listeners/udp_test.go

```diff
@@ -16,18 +16,28 @@ import (
 
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
+	"go.uber.org/fx"
 
-	"github.com/DataDog/datadog-agent/pkg/config"
-	"github.com/DataDog/datadog-agent/pkg/dogstatsd/packets"
+	"github.com/DataDog/datadog-agent/comp/core/config"
+	"github.com/DataDog/datadog-agent/comp/dogstatsd/packets"
+	"github.com/DataDog/datadog-agent/pkg/util/fxutil"
 )
 
-var (
-	packetPoolUDP        = packets.NewPool(config.Datadog.GetInt("dogstatsd_buffer_size"))
-	packetPoolManagerUDP = packets.NewPoolManager(packetPoolUDP)
-)
+func fulfillDepsWithConfig(t testing.TB, overrides map[string]interface{}) config.Component {
+	return fxutil.Test[config.Component](t, fx.Options(
+		config.MockModule,
+		fx.Replace(config.MockParams{Overrides: overrides}),
+	))
+}
```

Comment:
This seems like a useful helpers, should we move it to the config comp ?

Reply:
Ive been implementing `fulfillDeps*` helper functions in each set of tests that need the same set of dependencies fulfilled over and over again. Each helper function usually has as unique set of dependencies relevant to that test class. 

In this case, it's pretty generic as we only need the config (and overrides) with no extra options. 

The upside to making this more general (moving it into the config component to share with other tests) is it will save a few lines in each test class of implementing this same thing. 

The downside is: if you need to add another dependency to the tests that use this function, you will need to implement your own helper and stop using the share helper. 

So I can see it both ways. Let me know what you think.